src/formatDate.js

```diff
@@ -8,7 +8,46 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+
+  const dateInOldFormatArr = date.split(oldSeparator);
+  const dateInNewFormatArr = [];
+
+  const month = dateInOldFormatArr[fromFormat.indexOf('MM')];
+  const day = dateInOldFormatArr[fromFormat.indexOf('DD')];
+  let year;
+
+  if (date.length === 10) {
+    year = dateInOldFormatArr[fromFormat.indexOf('YYYY')];
+    year = `${year[2]}${year[3]}`;
+  } else if (date.length === 8) {
+    year = dateInOldFormatArr[fromFormat.indexOf('YY')];
+  }
+
+  for (const ch of toFormat) {
+    if (ch.includes('M')) {
+      dateInNewFormatArr.push(month);
+    }
+
+    if (ch.includes('D')) {
+      dateInNewFormatArr.push(day);
+    }
+
+    if (ch.includes('YYYY') && year < 30) {
```

Comment:
The condition `ch.includes('YYYY') && year < 30` is not sufficient to determine the correct century when converting from 'YY' to 'YYYY'. You should first check if the `fromFormat` includes 'YY' and then apply the conversion logic. Also, the variable `year` is a string, so comparing it with a number will not work as expected. You need to convert it to a number before making the comparison.